### PR TITLE
Enable TLS between Ironic and MariaDB

### PR DESCRIPTION
**Pull request by namnx228**
This PR is related to:
https://github.com/metal3-io/ironic-image/pull/206
https://github.com/metal3-io/metal3-dev-env/pull/525

### CODE REVIEW

**Review by maelk — APPROVED** [review]

File: tools/deploy.sh
Comment (maelk) [inline tools/deploy.sh:174]: maybe you can add MARIADB_HOST_IP to line 22 too, to fail early if it is unset

File: tools/deploy.sh
Comment (furkatgofurov7) [inline tools/deploy.sh:133]: You have to also update the documentation with new variables introduced [here](https://github.com/metal3-io/baremetal-operator/blob/master/docs/deploying.md#deployment-commands)

Comment (namnx228) [reply tools/deploy.sh:133]: Updated!. Thanks

File: tools/deploy.sh
Comment (namnx228) [reply tools/deploy.sh:174]: Fixed! Thanks

File: ironic-deployment/tls/default/tls.yaml
Comment (maelk) [inline ironic-deployment/tls/default/tls.yaml:46]: you need to mount the CA also in the ironic-api and ironic-conductor containers

File: ironic-deployment/tls/keepalived/tls.yaml
Comment (maelk) [inline ironic-deployment/tls/keepalived/tls.yaml:46]: same here

File: tools/run_local_ironic.sh
Comment (maelk) [inline tools/run_local_ironic.sh:16]: Please remove this change

**Review by maelk — CHANGES_REQUESTED** [review]

File: tools/run_local_ironic.sh
Comment (namnx228) [reply tools/run_local_ironic.sh]: Sorry, that was a mistake

File: ironic-deployment/tls/keepalived/tls.yaml
Comment (namnx228) [reply ironic-deployment/tls/keepalived/tls.yaml:52]: done!

File: ironic-deployment/tls/default/tls.yaml
Comment (namnx228) [reply ironic-deployment/tls/default/tls.yaml:52]: done!

**Review by maelk — APPROVED** [review]
/hold until ironic-image is merged

File: ironic-deployment/tls/default/tls.yaml
Comment (zaneb) [inline ironic-deployment/tls/default/tls.yaml:38]: ironic-inspector also uses the database.
We'll need a PR to add support for TLS in ironic-inspector-image before this can be merged also.

Comment (maelk) [reply ironic-deployment/tls/default/tls.yaml:38]: inspector uses sqlite afaik, so no changes there for now I would think

Comment (zaneb) [reply ironic-deployment/tls/default/tls.yaml:38]: Oh, you're right. Carry on then.

**Review by furkatgofurov7 — COMMENTED** [review]
Looks good to me. Please have another round of look.

/cc @zaneb @dhellmann @kashifest

**Pull request merged**
